packages/next-on-pages/templates/_worker.js/utils/fetch.ts

```diff
@@ -0,0 +1,40 @@
+export function patchFetchToAllowBundledAssets(): void {
+	const flagSymbol = Symbol.for('next-on-pages bundled assets fetch patch');
+
+	const alreadyPatched = (
+		globalThis.fetch as unknown as { [flagSymbol]: boolean }
+	)[flagSymbol];
+	if (alreadyPatched) {
+		return;
+	}
+
+	applyPatch();
+
+	(globalThis.fetch as unknown as { [flagSymbol]: boolean })[flagSymbol] = true;
+}
+
+function applyPatch() {
+	const originalFetch = globalThis.fetch;
+	globalThis.fetch = async (...args) => {
+		const request = new Request(...args);
+
+		if (request.url.startsWith('blob:')) {
+			try {
+				const url = new URL(request.url);
+				const binaryContent = (
+					await import(`./__next-on-pages-dist__/assets/${url.pathname}.bin`)
```

Comment:
Does `url.pathname` potentially need to be decoded? Check out what happens if the file has whitespace in it or whatever.

Reply:
interestingly that doesn't seem the case

at least for example I tried with whitespaces:
https://github.com/dario-piotrowicz/next-apps-for-testing/commit/a75ab62508c61cc5d3b604ca50d2929ccf89473b

and it seems to be working just fine:
https://6e9650dc.next-on-pages-test-5h3.pages.dev/api/og-custom

If I try to encode the url Next itself complains about it:
![Screenshot 2023-06-27 at 11 40 03](https://github.com/cloudflare/next-on-pages/assets/61631103/e1b5e866-d334-4351-8a2e-29f7b27a2996)

So I think we should be safe not decoding it 🙂
(and if this turns out to be needed we can always add the decoding later on)

Reply:
should be ok as discussed IRL 🙂